src/Tinfoil/Data/Signing.hs

```diff
@@ -31,30 +34,26 @@ parseSignatureVersion :: Text -> Maybe SignatureVersion
 parseSignatureVersion "v1" = pure SignatureV1
 parseSignatureVersion _ = Nothing
 
--- | Keyed-hash algorithm designator, for inclusion as a request parameter.
-data KeyedHashFunction =
-    HMAC_SHA256
-  deriving (Eq, Show, Enum, Bounded, Generic)
-
-instance NFData KeyedHashFunction
-
-renderKeyedHashFunction :: KeyedHashFunction -> Text
-renderKeyedHashFunction HMAC_SHA256 = "HMAC-SHA256"
-
-parseKeyedHashFunction :: Text -> Maybe KeyedHashFunction
-parseKeyedHashFunction "HMAC-SHA256" = pure HMAC_SHA256
-parseKeyedHashFunction _ = Nothing
-
-newtype SigningKey =
-  SigningKey {
-    unSigningKey :: ByteString
+newtype KeyId =
+  KeyId {
+    unKeyId :: ByteString
   } deriving (Eq, Show, Generic)
 
-instance NFData SigningKey
+instance NFData KeyId where rnf = genericRnf
 
 newtype RequestScope =
   RequestScope {
     unRequestScope :: ByteString
   } deriving (Eq, Show, Generic)
 
-instance NFData RequestScope
+instance NFData RequestScope where rnf = genericRnf
+
+newtype Signature =
+  Signature {
+    unSignature :: MAC
```

Comment:
Still nesting MAC type here? Maybe just wrapping bytestring be better to cover the symmetric / asymmetric cases? Not sure though.

I kind of feel like "API Signing" might be worth splitting out from "Signing" in general, constraints are different, and you don't get the properties I would expect from signing (i.e. no non-repudiation), given that this is pretty much raw MAC'ing maybe sticking with Signing to mean asymmetric signatures and Authenticating or MAC for the mac case works and the "API signing" terminology only needs to be worried about in zodiac.

Hope you don't mind all the name bike-shedding here, I think it is worth getting right though, and get the impression you agree from trying to split all this out nicely already.